ext/standard/array.c

```diff
@@ -6581,6 +6581,166 @@ PHP_FUNCTION(array_filter)
 }
 /* }}} */
 
+/* {{{ Internal function to find an array element for a user closure. */
+static zend_result php_array_find(const HashTable *array, zend_fcall_info fci, zend_fcall_info_cache fci_cache, zval *result_key, zval *result_value, bool negate_condition)
+{
+	zend_ulong num_key;
+	zend_string *str_key;
+	zval retval;
+	zval args[2];
+	zval *operand;
+
+	if (result_value != NULL) {
+  		ZVAL_UNDEF(result_value);
+	}
+
+	if (result_key != NULL) {
+  		ZVAL_UNDEF(result_key);
+	}
+
+	if (zend_hash_num_elements(array) == 0) {
+		return SUCCESS;
+	}
+
+	ZEND_ASSERT(ZEND_FCI_INITIALIZED(fci));
+
+	fci.retval = &retval;
+	fci.param_count = 2;
+	fci.params = args;
+
+	ZEND_HASH_FOREACH_KEY_VAL(array, num_key, str_key, operand) {
+		/* Set up the key */
+		if (!str_key) {
+			ZVAL_LONG(&args[1], num_key);
+		} else {
+			ZVAL_STR_COPY(&args[1], str_key);
+		}
+
+		ZVAL_COPY(&args[0], operand);
+
+		if (zend_call_function(&fci, &fci_cache) == SUCCESS) {
+			int retval_true;
+
+			retval_true = zend_is_true(&retval);
+			zval_ptr_dtor(&retval);
+
+			if (negate_condition) {
+				retval_true = !retval_true;
+			}
+
+			if (retval_true) {
+				if (result_value != NULL) {
+					ZVAL_COPY(result_value, &args[0]);
+				}
+
+				if (result_key != NULL) {
+					ZVAL_COPY(result_key, &args[1]);
+				}
+
+				zval_ptr_dtor(&args[0]);
+				zval_ptr_dtor(&args[1]);
+
+				return SUCCESS;
+			} else {
+				zval_ptr_dtor(&args[0]);
+				zval_ptr_dtor(&args[1]);
+			}
+		} else {
+			zval_ptr_dtor(&args[0]);
+			zval_ptr_dtor(&args[1]);
+
+			return FAILURE;
+		}
```

Comment:
This uses the suggestion of saving `result` to a variable to coalesce the dtor code:

```suggestion
			}
		}

		zval_ptr_dtor(&args[0]);
		zval_ptr_dtor(&args[1]);

		if (UNEXPECTED(result != SUCCESS)) {
			return FAILURE;
		}
```